.github/workflows/ci.yml

```diff
@@ -0,0 +1,134 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build and Check
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+permissions:
+  contents: read
+  security-events: write # upload Trivy Sarif results
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get Sources
+      uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: '1.22'
+        cache: false
+
+    - name: Tidy and verify modules
+      run: |
+        echo Tidying up modules
+        go mod tidy
+        echo Verifying modules
+        go mod verify
+
+    - name: Check formatting
+      run: test -z "$(go fmt $(go list ./...) | tee /dev/stderr)"
+
+    - name: Lint sources
+      uses: golangci/golangci-lint-action@v4
+
+    - name: Test
+      run: ./scripts/test
+
+  build-amd64:
+    runs-on: ubuntu-latest
+    needs:
+      - linter
+    steps:
+    - name: Get Sources
+      uses: actions/checkout@v3
+
+    - name: Determine Tag
+      id: get-TAG
+      run: |
+        export TAG=g$(git describe --always)
+        echo "TAG=$TAG"
+        echo "TAG=$TAG" >> "$GITHUB_ENV"
```

Comment:
I've seen a `./scripts/version` script that is extracting the `VERSION` in a different way. Should we use the same logic, or the same script, changing any Drone reference?

Reply:
I guess that you are referring to
```
DRONE_TAG:-$(git tag -l --contains HEAD | head -n 1)
```
This command reports the empty string when used in the GH action.
From Epinio I tried out `git describe --tags` first. This errored out.
In the end I was left with `git describe --always` to get something sensible (short commit id).